src/Tools/ExternalAccess/Xaml/External/XamlRequestContext.cs

```diff
@@ -0,0 +1,29 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using Microsoft.CodeAnalysis.LanguageServer;
+using Microsoft.CodeAnalysis.LanguageServer.Handler;
+using LSP = Microsoft.VisualStudio.LanguageServer.Protocol;
+
+namespace Microsoft.CodeAnalysis.ExternalAccess.Xaml;
+
+internal struct XamlRequestContext
+{
+    private readonly RequestContext _context;
+
+    public static XamlRequestContext FromRequestContext(RequestContext context)
+        => new(context);
+
+    private XamlRequestContext(RequestContext context)
+    {
+        _context = context;
+    }
+
+    public TextDocument? TextDocument => _context.TextDocument;
+
+    public LSP.ClientCapabilities ClientCapabilities => _context.GetRequiredClientCapabilities();
+
+    public T GetRequiredLspService<T>() where T : class, ILspService
+        => _context.GetRequiredLspService<T>();
```

Comment:
this worries me.  it exposes every service to xaml.  i woudl prefer to *not* do that.  we shoudl be specific about which services are exported, and wrap each in an EA wrapper.